### PR TITLE
Removing the unnecessary resource thats passed with query. 

### PR DESCRIPTION
**Pull request by georgethomas111**
Small Patch to correct an issue I found while using it. 

@johnhof @gsquire

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -138,6 +143,33 @@ func (c *CouchStore) GetStoreObj(jsonObj []byte) (error, caddyshack.StoreObject)
 	return nil, obj
 }
 
+func (c *CouchStore) MarshalStoreObjects(data []byte) (result []caddyshack.StoreObject, err error) {
```
Comment (gsquire) [inline main.go]: Can you return a slice of pointers here instead?

File: main.go [hunk 2]
```diff
@@ -138,6 +143,33 @@ func (c *CouchStore) GetStoreObj(jsonObj []byte) (error, caddyshack.StoreObject)
...
+
+	jsonStream := strings.NewReader(string(data))
+	jsonDecoder := json.NewDecoder(jsonStream)
```
Comment (gsquire) [inline main.go]: You can use a `bytes.NewBuffer` to avoid a copy.

File: main.go (hunk 1 above)
Comment (georgethomas111) [inline main.go]: Hey Actually its a slice of pointers thats returned. The caddyshack.StoreObject is an interface which implements a pointer. The getStoreObj actually creates the type from the type that's used to register with the store. So in essence its a slice of pointers.

File: main.go [hunk 3]
```diff
@@ -160,6 +192,31 @@ func (c *CouchStore) ReadOne(key string) (error, caddyshack.StoreObject) {
 	return err, obj
 }
 
+func (c *CouchStore) ReadOneFromView(desDocName string, viewName string, key string) (caddyshack.StoreObject, error) {
+
+	if strings.Contains(desDocName, "/") == false {
```
Comment (gsquire) [inline main.go]: `if !strings.Contains(desDocName, "/")`

File: main.go [hunk 4]
```diff
@@ -160,6 +192,31 @@ func (c *CouchStore) ReadOne(key string) (error, caddyshack.StoreObject) {
...
+		desDocName = "_design/" + desDocName
+	}
+	log.Print("Trying to read key " + key + " in viewName " + viewName + " of desDoc " + desDocName)
+	data, err := c.DbObj.GetView(desDocName, viewName, key)
+
+	if err != nil {
+		return nil, errors.New("Error retreiving : Key = " + key + " ViewName = " + viewName + " desDoc = " + desDocName + " " + err.Error())
```
Comment (gsquire) [inline main.go]: Prefer a format string here.

File: main.go [hunk 5]
```diff
@@ -160,6 +192,31 @@ func (c *CouchStore) ReadOne(key string) (error, caddyshack.StoreObject) {
...
+	} else {
+		// Print for now create store Object later.
+		// FIXME Handle unmarshalling over here.
+		result, err := c.MarshalStoreObjects(data)
+		if err != nil {
+			return nil, errors.New("Could not Marshal json" + err.Error())
+		}
+		if len(result) < 1 {
+			return nil, errors.New("caddyshack-couchdb : Key not found in database ")
+		} else {
```
Comment (gsquire) [inline main.go]: Get rid of this else and just return inside the block.

**Comment by georgethomas111** [comment]
@gsquire Made the changes and replied to one that was not possible to be make. Thanks for the comments.

File: main.go (hunk 1 above)
Comment (gsquire) [inline main.go]: Cool 👍

**Pull request merged**